refactor: 유저 도메인 SpringEvent -> RabbitMQ 로 전환 및 컨벤션 .md 파일로 관리

Mybread2 opened the pull request:
**유저 도메인 SpringEvent -> RabbitMQ 로 전환**
- 모든 곳에서 이벤트를 발행하는 건 안좋다고 판단하여 회원가입 / 회원탈퇴 / 팔로우 / 사용자 평가 총 4가지 경우에만 이벤트 발행
- 패키지 구조를 어떤 식으로 가져가야할지 고민해보겠습니다.
- 인증 도메인 쪽에서 별도의 DLX / DLQ / SimpleRabbitListenerContainerFactory 설정 필요

**RabbitMQ 네이밍 컨벤션 및 패키지 구조 문서화**
- Exchange, Routing Key, Queue 명명 규칙 상세 정리
- Java 상수 관리 가이드 추가
- 이벤트 DTO 글로벌 패키지 배치 및 구조 설명 추가
- 패키지 구조 예시 다이어그램 포함
- Queue 이름 상수 관리 필요성 및 위치 가이드 포함
- RabbitMQ Config 위치에 대한 권장사항 간단 언급

Mybread2 committed:
refactor: 유저 도메인 SpringEvent -> RabbitMQ 로 전환

- 모든 곳에서 이벤트를 발행하는 건 안좋다고 판단하여 회원가입 / 회원탈퇴 / 팔로우 / 사용자 평가 총 4가지 경우에만 이벤트 발행
- 우선 인증 쪽에서는 유저의 모든 이벤트를 받도록 설정해놨는데 추후에 수정 예정
- 아직 패키지 구조를 어떤 식으로 가져가야할지 고민

Mybread2 committed:
refactor: 명시적으로 직렬화된 JSON에 클래스 타입 정보를 포함

- 역직렬화 시 Jackson이 자동으로 실제 타입으로 복원해줘서 Object data를 안전하게 처리

Mybread2 committed:
refactor: AuthRabbitMQConfig 클래스에서 의존성 없애기 위해 하드코딩

Mybread2 committed:
Merge branch 'dev-v3' of https://github.com/momo-spring-project/momo-backend into feat/AWS-v3

Mybread2 committed:
Merge branch 'dev-v3' of https://github.com/momo-spring-project/momo-backend into feat/AWS-v3

Mybread2 committed:
feat: .gitignore 설정

Mybread2 committed:
refactor: 인증 도메인에서 필요한 이벤트용 dto 설정
- 지금 당장은 회원탈퇴에만 영향을 받아서 필요한 것만 남겨둠

hojin915 reviewed (approved):
확인했습니다
패키지 구조 지금 PR 올리신 이 방식대로 따라하면 될까요?
디렉토리를 하나 더 만든다고도 들었던 것 같은데 정확히 어떤방식인지 기억이 안납니다

Mybread2 committed:
docs: RabbitMQ 네이밍 컨벤션 및 패키지 구조 문서화

- Exchange, Routing Key, Queue 명명 규칙 상세 정리
- Java 상수 관리 가이드 추가
- 이벤트 DTO 글로벌 패키지 배치 및 구조 설명 추가
- 패키지 구조 예시 다이어그램 포함
- Queue 이름 상수 관리 필요성 및 위치 가이드 포함
- RabbitMQ Config 위치에 대한 권장사항 간단 언급

팀 내 메시징 시스템 설계 표준 공유용 문서 작성

Mybread2 committed:
fix: README 용으로 코드 수정

Mybread2 commented:
제가 .md 파일 만들어서 컨벤션 올려놨습니다! 
그리고 우선 유저랑 인증 도메인, 그리고 global 쪽 패키지 구조를 잡아놨습니다. 이를 참고하셔서 해주시면 될거같습니다!
혹시 수정할 부분 있으면 알려주세요! 제가 잘못 인지하고 있을 수도 있어서 알려주시면 좋을거같습니다.

Mybread2 committed:
fix: rabbitmq-naming-convention.md 프로젝트 구조에 맞게 수정

ko-dongwon reviewed (approved):
확인했습니다! 고생하셨습니다:)

Zyooon reviewed (approved):
확인했습니다.

ZeroColaa commented:
확인했습니다 고생하셨습니다!
